### ISSUE TITLE
Indicators don't automatically show, have to toggle extension

### ISSUE THREAD

**Issue by fgblomqvist** [ubuntu/gnome-shell-extension-appindicator#107]
For some reason it seems like this extension is not continuously monitoring stuff, because if I launch an application after this extension has been turned on, it will not show up in the tray. In order for it to show up, I have to toggle the extension on and off. Pretty annoying since I have to do that on every boot, and multiple times throughout the day. Is there a log for the extension?

I am running GNOME 3.26 on Ubuntu 17.10 and have the latest extension version from the extensions website.

EDIT:
Found the issue in the logs (`journalctl /usr/bin/gnome-shell -f`):
`[AppIndicatorSupport-WARN] Failed to acquire org.kde.StatusNotifierWatcher`

**Comment by jhasse** [comment]
Are you using the Ubuntu session or the vanilla GNOME session?

**Comment by fgblomqvist** [comment]
I'm using the GNOME on Xorg session?

**Comment by fgblomqvist** [comment]
Toggling no longer works. I have to restart the shell (Alt + F2 -> r) for it to refresh. It seems like the unowning of the name never happens, and therefore the NameWatchdog vanished handler never fires.

**Comment by fgblomqvist** [comment]
Using d-feet, it looks like `indicator-application-service` owns the name, and is therefore blocking this extension.

**Comment by fgblomqvist** [comment]
I'm gonna put this here in case other people come here because the extension doesn't fully work:
I went ahead and uninstalled `indicator-application' (`sudo apt remove indicator-application`) and disabled this extension. Now all icons are handled by TopIcons Plus again (like they were in GNOME 3.24). Everything works really smoothly, appearing and disappearing like they should. Since the kde.org.StatusNotifierWatcher dbus is now gone, it might cause breakage in the future (if an app requires that dbus), but I will just keep it uninstalled until this mess has been figured out (and either received better support or better docs, wouldn't mind getting down and dirty with this stuff if only I could figure out what is what).

**Comment by LeDucDuBleuet** [comment]
Thank you @fgblomqvist! Removing the indicator application solved it for me as well and it is even working with the original TopIcons extension with Nextcloud client, Tranmission, Shutter and KeePassXC in my case. I do not notice any undesirable side-effect. I wonder what this indicator-application package is supposed to do? Is it really safe to remove it?

**Comment by fgblomqvist** [comment]
@LeDucDuBleuet Like I said, it probably just implements the new interface they wanted to switch to. It certainly doesn't seem to cause breakage right now, but might in the future.

**Comment by jhasse** [comment]
So this seems to be related to the Ubuntu package `indicator-application` and not this extension?

**Comment by fgblomqvist** [comment]
Could be I suppose? Although, I don't see the need for this extension if TopIcons Plus already does everything this one does and more. I think someone needs to clarify what indicator-application is exactly, and why this extension is needed.

**Comment by jhasse** [comment]
TopIcons Plus has a different (older) technical implementation. This extension uses a DBus API which should result in less bugs.

For example: With TopIcons Plus, if you right click on a tray icon the application draws the menu itself. It will also grab the keyboard to enable selecting entries. When you press <kbd>esc</kbd> to close the menu, this is actually handled by the application, not by your DE. This means that a bug in the app could cause your whole desktop to hang (or rather: your keyboard and mouse input not having any effect).

To prevent this, the idea is to isolate what applications can do. Especially that "grabbing the whole keyboard and mouse" shouldn't be allowed. This is also what Wayland tries to enforce, that's why TopIcons Plus doesn't work on Wayland (only via Xwayland).


Please reopen, if you think there's something this extension could do to workaround the `indicator-application` package being installed.

**Comment by DominicWatson** [comment]
I was just debugging this and came across the same error in the logs re being unable to acquire DBUS WATCHER NAME. Playing around, I tried renaming the watcher to something random (and renamed other references to it) - the message went away (but the watcher didn't work either ;) ).

e.g. 

    const WATCHER_BUS_NAME = KDE_PREFIX + '.StatusNotifierWatcherForAppIndicators';

I reverted my experiment.

Uninstalling other extensions seemed to fix the issue and I felt like I was on some kind of track. But then I re-enabled them to restart some more debugging and all of a sudden everything was working again.

I'm adding this in the slim chance that it reveals something you've not seen that might lead to some revelation. Clearly I don't understand what is happening under the hood of:

        this._ownName = Gio.DBus.session.own_name(WATCHER_BUS_NAME,
                                  Gio.BusNameOwnerFlags.NONE,
                                  Lang.bind(this, this._acquiredName),
                                  Lang.bind(this, this._lostName));

Some kind of proxy to Gnome DBUS?? meaning that the bug is somewhere in there? Is the real name of the BUS important and does it have to be unique? Or could they be anything?

Anyways, feel free to ignore this - but, again, just posting in case it triggers anything.
